skalenetwork/filestorage.js
# Docs Development Automation Added

Reason for Update
---
Adds "hot reload/watch" functionality when updating documentation



**What**:

- Added submodule doc-ui to root which calls: https://github.com/skalenetwork/doc-ui
- Added doc-utils as npm package (GitHub Package) -> https://github.com/skalenetwork/doc-utils
- Added local-playbook-ui.yml -> Core Hot Reload Template for Live Doc Edits
- Migrated from NPM to Yarn to simplify package usage [and results in faster builds]



**Why**:

- These changes make it easier to see changes happen to the docs in the style of the SKALE documentation suite. Improved process results in more updates more frequently.



**How**:

- Using a combination of the doc-utils package as well as some minor updates to the ESLint config to avoid gulp build errors




**Checklist**:






-   [ ] Tests N/A

-   [] Ready to be merged



Yes, this will not impact the core repo meaning it does not need to pass the tests above.



How to Test
---

- Checkout this PR, and setup as normal ```yarn```
- Run ```yarn docs:watch``` and wait for the UI to build, once built opening ```127.0.0.1:444``` in your browser should allow you to select the filestorage.js folder, you must manually click into the file
- After manually navigating in the browser changes will occur in the browser automatically when changes are made in the **docs** folder.
